crates/rustc_codegen_spirv/src/builder/builder_methods.rs

```diff
@@ -917,9 +920,47 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             s1
         }
     }
+
+    // HACK(eddyb) helper shared by `typed_alloca` and `alloca`.
+    fn declare_func_local_var(
+        &mut self,
+        ty: <Self as BackendTypes>::Type,
+        _align: Align,
+    ) -> SpirvValue {
+        let ptr_ty = self.type_ptr_to(ty);
+
+        // "All OpVariable instructions in a function must be the first instructions in the first block."
+        let mut builder = self.emit();
+        builder.select_block(Some(0)).unwrap();
+        let index = {
+            let block = &builder.module_ref().functions[builder.selected_function().unwrap()]
+                .blocks[builder.selected_block().unwrap()];
+            block
+                .instructions
+                .iter()
+                .enumerate()
+                .find_map(|(index, inst)| {
+                    if inst.class.opcode != Op::Variable {
+                        Some(InsertPoint::FromBegin(index))
+                    } else {
+                        None
+                    }
+                })
+                .unwrap_or(InsertPoint::End)
+        };
+        // TODO: rspirv doesn't have insert_variable function
```

Comment:
Do we need a task on rspirv? If so we should file and link here

Reply:
This stuff is an old comment that can probably removed (expecting high-level APIs from `rspirv` is counterproductive).